src/management/tasks/promotion/promotion-task.component.ts

```diff
@@ -14,15 +14,62 @@
  * limitations under the License.
  */
 import { PromotionTask } from '../../../entities/task/task';
+import { IPromise } from 'angular';
+import { PromotionService } from '../../../services/promotion.service';
 
 class PromotionTaskComponentController {
+  constructor(private readonly $mdDialog: angular.material.IDialogService, private readonly promotionService: PromotionService) {
+    'ngInject';
+  }
+
   public task: PromotionTask;
+  public taskHandled: () => void;
+
+  public openConfirmationDialog(acceptPromotion: boolean): IPromise<void> {
+    const dialogResult = acceptPromotion ? this.openAcceptationDialog() : this.openRejectionDialog();
+
+    return dialogResult.then((dialogValidation) => {
+      if (dialogValidation) {
+        this.promotionService.processPromotion(this.task.data.promotionId, acceptPromotion).then(() => this.taskHandled());
+      }
+    });
+  }
```

Comment:
maybe we can handle the problem from the other side

instead of calling this from the template with a boolean maybe you can directly call `openAcceptationDialog` and `openRejectionDialog()` in the template and externalize the return block in a private method ?

it should be more readable from the template pov
wdyt ?